### PR TITLE
backwards compatibility fixes

### PR DESCRIPTION
**Pull request by cristipogacean**
**Changes:**
* GetConfiguredEndpoints result payload contains "endpoints" object 
* GetConfiguredNodesOnEndpoints - result payload contains now "opcNodes" object
* UnpublishAllNodes - not providing an endpoint results into removing completely the configuration
* E2E tests adapted accordingly

*Note*: Documentation will be updated in a separate PR

### CODE REVIEW

File: api/src/Microsoft.Azure.IIoT.OpcUa.Api.Publisher/src/Models/GetConfiguredEndpointsResponseApiModel.cs [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    using System.Collections.Generic;
+
+    /// <summary>
+    /// Result of GetConfiguredEndpoints method call
+    /// </summary>
+    [DataContract]
+    public class GetConfiguredEndpointsResponseApiModel {
+
+        /// <summary>
+        /// Service result in case of error
```
Comment (dacolgit) [inline api/src/Microsoft.Azure.IIoT.OpcUa.Api.Publisher/src/Models/GetConfiguredEndpointsResponseApiModel.cs]: >Service result in case of error [](http://example.com/codeflow?start=12&length=31)

the comment seems not related to the property

File: api/src/Microsoft.Azure.IIoT.OpcUa.Api.Publisher/src/Models/GetConfiguredNodesOnEndpointResponseApiModel.cs [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    using System.Collections.Generic;
+
+    /// <summary>
+    /// Result of GetConfiguredNodesOnEndpoint method call
+    /// </summary>
+    [DataContract]
+    public class GetConfiguredNodesOnEndpointResponseApiModel {
+
+        /// <summary>
+        /// Service result in case of error
```
Comment (dacolgit) [inline api/src/Microsoft.Azure.IIoT.OpcUa.Api.Publisher/src/Models/GetConfiguredNodesOnEndpointResponseApiModel.cs]: >Service result in case of error [](http://example.com/codeflow?start=12&length=31)

the comment seems not related to the property

File: api/src/Microsoft.Azure.IIoT.OpcUa.Api.Publisher/src/Models/GetConfiguredEndpointsResponseApiModel.cs (hunk 1 above)
Comment (cristipogacean) [reply api/src/Microsoft.Azure.IIoT.OpcUa.Api.Publisher/src/Models/GetConfiguredEndpointsResponseApiModel.cs]: thanks!

File: api/src/Microsoft.Azure.IIoT.OpcUa.Api.Publisher/src/Models/GetConfiguredNodesOnEndpointResponseApiModel.cs (hunk 2 above)
Comment (cristipogacean) [reply api/src/Microsoft.Azure.IIoT.OpcUa.Api.Publisher/src/Models/GetConfiguredNodesOnEndpointResponseApiModel.cs]: thanks!

File: components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa/src/Publisher/Extensions/PublishedNodesEntryModelEx.cs [hunk 3]
```diff
@@ -28,7 +28,7 @@ public static class PublishedNodesEntryModelEx {
                 return false;
             }
 
-            if (string.Compare(model.EndpointUrl.OriginalString, that.EndpointUrl.OriginalString, StringComparison.OrdinalIgnoreCase) != 0) {
+            if (model.EndpointUrl != that.EndpointUrl) {
```
Comment (karok2m) [inline components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa/src/Publisher/Extensions/PublishedNodesEntryModelEx.cs:31]: Why make comparison case-sensitive ? URL comparison should in general be case-insensitive.

Comment (cristipogacean) [reply components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa/src/Publisher/Extensions/PublishedNodesEntryModelEx.cs:31]: I'm comparing now the **Uri** EndpointUrl objects no longer the strings.

**Review by dacolgit — APPROVED** [review]

File: components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa/src/Publisher/Extensions/PublishedNodesEntryModelEx.cs (hunk 3 above)
Comment (karok2m) [reply components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa/src/Publisher/Extensions/PublishedNodesEntryModelEx.cs:31]: Ah, right. Would this also ignore trailing `/` characters in URIs ?

Comment (cristipogacean) [reply components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa/src/Publisher/Extensions/PublishedNodesEntryModelEx.cs:31]: yes, trailing `/` are ignored, among others...

File: components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/src/Engine/StandaloneJobOrchestrator.cs [hunk 4]
```diff
@@ -786,82 +786,100 @@ IJsonSerializer jsonSerializer
...
-                finally {
-                    _lockJobs.Release();
-                }
+                else {
 
-                if (!found) {
-                    throw new MethodCallStatusException((int)HttpStatusCode.NotFound,
-                        $"Endpoint not found: {request.EndpointUrl}");
+                    // Remove all entries
+                    _publishedNodesEntries.Clear();
```
Comment (karok2m) [inline components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/src/Engine/StandaloneJobOrchestrator.cs:872]: We should also update `_lastKnownFileHash ` like this together with `_publishedNodesEntries.Clear();`:

```
_publishedNodesEntries.Clear();
_lastKnownFileHash = string.Empty;
```

File: components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/src/Engine/StandaloneJobOrchestrator.cs [hunk 5]
```diff
@@ -786,82 +786,100 @@ IJsonSerializer jsonSerializer
             CancellationToken ct) {
             _logger.Information("{nameof} method triggered", nameof(UnpublishAllNodesAsync));
             var sw = Stopwatch.StartNew();
+            // when no endpoint is specified remove all the configuration
+            var purge = null == request.EndpointUrl;
```
Comment (karok2m) [inline components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/src/Engine/StandaloneJobOrchestrator.cs:790]: Let's document that this behavior is coming from `2.5.x` and we have it here for backward-compatibility.

File: components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/src/Engine/StandaloneJobOrchestrator.cs (hunk 4 above)
Comment (karok2m) [reply components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/src/Engine/StandaloneJobOrchestrator.cs:872]: `PersistPublishedNodes()` does this already.

**Review by karok2m — APPROVED** [review]

File: components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/src/Engine/StandaloneJobOrchestrator.cs (hunk 5 above)
Comment (karok2m) [reply components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/src/Engine/StandaloneJobOrchestrator.cs:790]: Note about backward compatibility added in #1596.
